### PR TITLE
feat: notifications tracking and reporting

### PR DESCRIPTION
**Pull request by vbuch**
Phase 1 — Data model & indexes:
- Add clickedAt and openedAt fields to NotificationMatch type (ingest + web)
- Add MongoDB compound indexes: (userId, clickedAt) and (userId, openedAt)
- Add Firestore composite indexes for analytics query patterns

Phase 2 — Click and open tracking:
- notification-sender: notification URLs now point to /notification?id=<matchId>
  instead of /m/<messageId>, enabling server-side click tracking via redirect
- New page /notification: records clickedAt (first-write-wins, matchId as token) and redirects to /?messageId=<messageId>; no auth required
- SW template: simplified notificationclick handler — navigate existing window to the redirect URL (instead of postMessage + fire-and-forget fetch fallback)
- Regenerate web/public/firebase-messaging-sw.js from updated template
- Update mark-read route to also set openedAt on first open (first-write-wins) while keeping existing readAt behaviour for unread badge UX

Phase 3 — Analytics API:
- Add pure aggregation module (aggregation.ts) with: aggregateNotificationKpis, getMessageIdsForMode, extractHeatmapPointsFromMessage, buildHeatmapResult
- Add GET /api/notifications/report?mode=all|clicked|opened: KPIs, heatmap points from triggering message coordinates, per-source breakdown; privacy threshold of 50 records; gated by hasReportPagesEnabled(); node runtime with caching headers

Phase 4 — Report page:
- Footer: add Известия link when report pages are enabled
- ClientLayout / layout.tsx: thread showNotificationsReportLink prop
- New page /notifications-report with gating, Bulgarian UI copy
- NotificationsReportClient: KPI cards, mode filter, source table, heatmap section with privacy-threshold message
- NotificationsReportMapClient: Leaflet heatmap renderer reusing HistoryMapClient architecture

Phase 5 — Tests (41 new test cases):
- aggregation.test.ts: 19 unit tests for all pure functions
- notification/page.test.ts: 6 tests for redirect page (click recording, idempotency, missing/unknown ID fallback, URL encoding)
- report/route.test.ts: 10 tests covering KPIs, filter modes, privacy threshold, source breakdown, 503 gating, error handling
- mark-read/route.test.ts: extended with 2 tests for openedAt behaviour

### CODE REVIEW

File: ingest/notifications/notification-sender.ts [hunk 1]
```diff
@@ -120,8 +120,13 @@ export function buildNotificationPayload(
     ? ` (${Math.round(match.distance)}m от вашия район)`
     : "";
 
-  // Message ID is URL-encoded to handle any special characters
-  const messageUrl = `${APP_URL}/m/${encodeURIComponent(match.messageId)}`;
+  // Use match ID in the notification URL for server-side click tracking.
+  // The /notification page records clickedAt then redirects to the message.
+  // Fall back to the direct message URL if match.id is not set (should not
+  // happen after notify, but guards against edge cases in tests/backfill).
+  const messageUrl = match.id
```
Comment (vbuch) [inline ingest/notifications/notification-sender.ts:127]: I like the shorter shape of `/m/<id>`. Let's have the notification one similar - `/n/<id>`

Also... are we sure matchIs are not a huge string that will cause issues?

File: plan.md
Comment (vbuch) [inline plan.md:1]: Great that we have the plan.md as something to review, but let's not merge it in main.
